### PR TITLE
[2.7] Issue 543846: Add DB2z support for named parameters with stored proce…

### PR DESCRIPTION
**Pull request by dazey3**
for #362

Signed-off-by: Will Dazey <dazeydev.3@gmail.com>

### CODE REVIEW

File: foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/databaseaccess/BindCallCustomParameter.java [hunk 1]
```diff
@@ -71,6 +73,10 @@ public void set(DatabasePlatform platform, PreparedStatement statement, int inde
         platform.setParameterValueInDatabaseCall(obj, statement, index, session);
     }
 
+    public void set(DatabasePlatform platform, CallableStatement statement, String name, AbstractSession session) throws SQLException {
```
Comment (lukasj) [inline foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/databaseaccess/BindCallCustomParameter.java]: would be good to add some javadoc here

File: foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/databaseaccess/InOutputParameterForCallableStatement.java [hunk 2]
```diff
@@ -51,11 +52,18 @@ public InOutputParameterForCallableStatement(Object inParameter, DatabaseField o
...
     }
 
+    @Override
     public void set(DatabasePlatform platform, PreparedStatement statement, int index, AbstractSession session) throws SQLException {
         platform.setParameterValueInDatabaseCall(inParameter, statement, index, session);
         super.set(platform, statement, index, session);
     }
 
+    @Override
+    public void set(DatabasePlatform platform, CallableStatement statement, String index, AbstractSession session) throws SQLException {
```
Comment (lukasj) [inline foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/databaseaccess/InOutputParameterForCallableStatement.java]: calling String arg here `index` looks confusing to me

**Comment by dazey3** [comment]
@lukasj 
>One thing which is not clear to me - new methods do take AbstractSession as an argument and the only usage I've found is to log an exception. If this is correct, is this really needed? Cannot global AbstractSessionLog.getLog be used or some kind of recoverable exception be thrown instead?

Correct, I passed AbstractSession down the call chain so that I can log warnings. Also, there was already many existing calls in the INTERNAL api that pass AbstractSession down, so I figured it was fine to add. I didn't know a static call like `AbstractSessionLog.getLog()` was available! I updated the PR to reflect this change.

File: foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/databaseaccess/InOutputParameterForCallableStatement.java (hunk 2 above)
Comment (dazey3) [reply foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/databaseaccess/InOutputParameterForCallableStatement.java]: I agree, ty for noticing this. Ill go back through and change the var names to be "int parameterIndex" and "String parameterName". Does that sound good?

File: foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/databaseaccess/BindCallCustomParameter.java (hunk 1 above)
Comment (dazey3) [reply foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/databaseaccess/BindCallCustomParameter.java]: Added :) thanks!

**Pull request merged**